app/models/katello/concerns/operatingsystem_extensions.rb

```diff
@@ -11,28 +11,91 @@ module OperatingsystemExtensions
         before_create :set_atomic_attributes, :if => proc { |os| os.name == ::Operatingsystem::REDHAT_ATOMIC_HOST_OS }
       end
 
-      def assign_templates!
-        # Automatically assign default templates
-        if self.family == 'Redhat'
-          TemplateKind.all.each do |kind|
-            if name == ::Operatingsystem::REDHAT_ATOMIC_HOST_OS && kind.name == "provision"
-              provisioning_template_name = Setting["katello_default_atomic_provision"]
-            else
-              provisioning_template_name = Setting["katello_default_#{kind.name}"]
-            end
+      def find_related_os(ignored_os)
+        [true, false].each do |with_major|
+          query = Operatingsystem.where(name: self.name).where.not(id: ignored_os)
+          query = query.where(major: self.major) if with_major
+          related_os = query.order(created_at: :desc).last
```

Comment:
Sorry, my suggestion wasn't correct at all. :see_no_evil:  I wanted to suggest to use the highest minor in case we filter for major version. And this could also be applied to the case without major version (There we could filter for the highest major / minor).
But I agree, this would make it more complex as it currently is and also using the latest created OS is a valid approach.